### ISSUE TITLE
Custom http headers set in WebViewHandler are not sent

### ISSUE THREAD

**Issue by programatix** [dotnet/maui#12206]
### Description

Custom headers set in WebViewHandler are not sent. On my test, both Android and iOS have this issue, so I believe somehow, MAUI subsystem overwritten the headers.

I have prepared a sample solution at https://github.com/programatix/MauiWebViewBug. The test solution provided opens the URL https://manytools.org/http-html-text/http-request-headers which will echo the request headers.

The sample sets the following headers
* "Authorization": "bearer hello"
* "Accept-Language": "US"
* "MyHeader": "there"

###### MAUI result
![image](https://user-images.githubusercontent.com/3293865/208389897-28cafa71-df98-4483-946f-e061de73c0d2.png)
![image](https://user-images.githubusercontent.com/3293865/208389991-8c40aa5f-0bc6-4d9e-957d-94ba5a8a3f97.png)

###### Xamarin result
I also included the Xamarin sample and the result is correct.
![image](https://user-images.githubusercontent.com/3293865/208389646-4385f746-bb48-4f24-ac9a-14746af5c30e.png)

### Steps to Reproduce

1. Open the sample solution
2. Run

### Link to public reproduction project repository

https://github.com/programatix/MauiWebViewBug

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, I was *not* able test on other platforms

### Affected platform versions

Android 13, iOS 16.1.1

**Comment by SarthakGz** [comment]
This is really important for our application as well please look into this

**Comment by programatix** [comment]
I found a workaround on this by overriding ```CreatePlatformView()``` in the handlers. The original ```MauiWebView``` (Android) and ```MauiWKWebView``` (iOS) somehow overwritten the headers loaded by our ```LoadUrl``` call.

#### Android
```
        protected override global::Android.Webkit.WebView CreatePlatformView()
        {
            var view = new global::Android.Webkit.WebView(Context!)
            {
                LayoutParameters = new LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent)
            };

            view.Settings.JavaScriptEnabled = true;
            view.Settings.DomStorageEnabled = true;
            view.Settings.SetSupportMultipleWindows(true);

            return view;
        }
```

#### iOS
```
        protected override WKWebView CreatePlatformView()
        {
            return new WKWebView(CoreGraphics.CGRect.Empty, new WKWebViewConfiguration());
        }
```

I also found something weird with MAUI WebView control. If I ignore the ```Source``` property and introduced a user defined property in ```MyWebView```, and later use that in the handlers to perform ```LoadUrl()``` (Android) or ```LoadRequest()``` (iOS), the WebView will be blank. This is also true after I use the workaround shown above. I guess the page is loaded but not rendered by the WebView control?

**Comment by rbakhshi** [comment]
@programatix are you sure this ever worked in Xamarin? We had a similar need and ended up intercepting the request and injecting the headers there.

Basically:
1. Override `ShouldInterceptRequest` 
2. If request already has the header, return `base.ShouldInterceptRequest`
3. if not, then
   1. call `LoadUrl` on the native view with original URL and additional headers
   2. return and empty `WebResourceResponse` for the current request.

Not very elegant but that was the only way it worked on Xamarin.

**Comment by programatix** [comment]
@rbakhshi Yes, it works in Xamarin and I have been using it since last year. The sample solution I provided has Xamarin version in it.

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 2.0. Can repro on iOS and android platform with sample project.
[MauiWebViewBug.zip](https://github.com/dotnet/maui/files/11998895/MauiWebViewBug.zip)
![12206](https://github.com/dotnet/maui/assets/77434865/ed894f2b-11f2-456e-99b3-c241291b8469)

**Comment by manicoder** [comment]
any update on this i am also facing the same issue

**Comment by mnel75** [comment]
News about the issue ?

**Comment by programatix** [comment]
Issue is still there for .NET 7.0. For .NET 8.0, I can't test it since my macbook doesn't support Ventura.
